lab_1_classify_by_unigrams/main.py

```diff
@@ -4,51 +4,123 @@
 """
 
 
-def tokenize(text: str) -> list[str] | None:
-    """
-    Splits a text into tokens, converts the tokens into lowercase,
-    removes punctuation, digits and other symbols
-    :param text: a text
-    :return: a list of lower-cased tokens without punctuation
+def tokenize(text: str) -> list[str]:
     """
+        Splits a text into tokens, converts the tokens into lowercase,
+        removes punctuation, digits and other symbols
+        :param text: a text
+        :return: a list of lower-cased tokens without punctuation
+        """
+    if type(text) != str:
+        return None
+    text = text.lower()
+    for i in text:
+        if not i.isalpha() and i != ' ':
+            text = text.replace(i, '')
+    for i in text:
+        if i == ' ':
+            text = text.replace(' ','')
+    tokens = list(text)
+    return tokens
 
 
-def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
-    """
-    Calculates frequencies of given tokens
-    :param tokens: a list of tokens
-    :return: a dictionary with frequencies
+
+def calculate_frequencies(tokens : list[str] | None) -> dict[str, float] | None:
     """
+       Calculates frequencies of given tokens
+       :param tokens: a list of tokens
+       :return: a dictionary with frequencies
+       """
+    if type(tokens) != list:
+        return None
+    for i in tokens:
+        if type(i) != str:
+            return None
+    dictionary = {}
+    length = len(tokens)
+    for i in tokens:
+        if i in dictionary:
+            dictionary[i] += 1/length
+        else:
+            dictionary[i] = 1/length
+    return dictionary
+
+
+
+
 
 
 def create_language_profile(language: str, text: str) -> dict[str, str | dict[str, float]] | None:
     """
-    Creates a language profile
-    :param language: a language
-    :param text: a text
-    :return: a dictionary with two keys – name, freq
-    """
-
+      Creates a language profile
+      :param language: a language
+      :param text: a text
+      :return: a dictionary with two keys – name, freq
+      """
+    if type(language) != str or type(text) != str:
+        return None
+    new_dictionary = {}
+    text = tokenize(text)
+    freq = calculate_frequencies(text)
+    new_dictionary[language] = freq
+    return new_dictionary
```

Comment:
Does your produced dictionary correspond to the one described in Step 3 of the documentation?

Reply:
Fixed